packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts

```diff
@@ -249,6 +250,41 @@ class DescopeWc extends BaseDescopeWc {
     return Promise.race([promises, toPromise]);
   }
 
+  get isDismissScreenErrorOnInput() {
+    return this.getAttribute('dismiss-screen-error-on-input') === 'true';
+  }
+
+  #handleGlobalErrors({
+    errorText,
+    errorType,
+  }: {
+    errorText: string;
+    errorType: string;
+  }) {
+    const updateGlobalError = () => {
+      let transformedErrorText = errorText;
+      try {
+        transformedErrorText =
+          this.errorTransformer?.({
+            text: errorText,
+            type: errorType,
+          }) || errorText;
+      } catch (e) {
+        this.loggerWrapper.error('Error transforming error message', e.message);
+      }
+      replaceElementMessage(
+        this.contentRootElement,
+        'error-message',
+        transformedErrorText,
+      );
+    };
+
+    // we do not know if the page is going to be updated or not,
+    // so we are updating the error message component before and after the screen update
+    this.addEventListener('screen-updated', updateGlobalError, { once: true });
```

Comment:
when does screen-update event is fired? in custom screens only?

Reply:
When rendering a flow screen or a custom screen